### PR TITLE
Refactor torch.onnx documentation

### LINKED ISSUE #108274: API: Restructure ONNX API doc with clear separation between TorchScript-based and FX-based explorer

### PR DESCRIPTION
**Pull request by thiagocrepaldi**
Pull Request resolved: https://github.com/pytorch/pytorch/pull/108379
Fixes #108274

Approved by @wschin @justinchuby

### CODE REVIEW

File: docs/source/onnx_dynamo.rst [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+===============================
+
+.. automodule:: torch.onnx
+  :noindex:
+
+.. contents:: :local:
+    :depth: 3
+
+.. warning::
+  The ONNX exporter for TorchDynamo is under active development and is subject to rapid change.
```
Comment (abock) [inline docs/source/onnx_dynamo.rst:11]: > The ONNX exporter for TorchDynamo is a rapidly evolving beta technology.

File: docs/source/onnx_dynamo.rst [hunk 2]
```diff
@@ -0,0 +1,156 @@
...
+
+  model = MLPModel()
+  tensor_x = torch.rand((97, 8), dtype=torch.float32)
+  export_output = torch.onnx.dynamo_export(model, tensor_x)
+
+As the code above shows, all you need is to provide :func:`torch.onnx.dynamo_export` with an instance of the model and its input.
+The exporter will then return an instance of :class:`torch.onnx.ExportOutput` that contains the exported ONNX graph along with extra information.
+
+The in-memory model available through ``export_output.model_proto`` is an ``onnx.ModelProto`` object in compliance with the `ONNX IR spec <https://github.com/onnx/onnx/blob/main/docs/IR.md>`_.
+The ONNX model be serialized into a `Protobuf file <https://protobuf.dev/>`_ using the :meth:`torch.onnx.ExportOutput.save` API.
```
Comment (abock) [inline docs/source/onnx_dynamo.rst:83]: > The ONNX model may then be serialized ...

File: docs/source/onnx_dynamo_onnxruntime_backend.rst [hunk 3]
```diff
@@ -0,0 +1,10 @@
+ONNX Backend for TorchDynamo
+============================
+
+For a quick overview of ``torch.compiler``, see :ref:`torch.compiler_overview`.
+
+.. warning::
+  The ONNX backend for TorchDynamo is under active development and is
```
Comment (abock) [inline docs/source/onnx_dynamo_onnxruntime_backend.rst:7]: > The ONNX backend for `torch.compile` is a rapidly evolving beta technology.

File: docs/source/onnx_dynamo_onnxruntime_backend.rst [hunk 4]
```diff
@@ -0,0 +1,10 @@
...
+  subject to rapid change.
+
+.. autofunction:: torch.onnx.is_onnxrt_backend_supported
```
Comment (abock) [inline docs/source/onnx_dynamo_onnxruntime_backend.rst:10]: Would be good to replicate the same demo you have for the exporter and invoke it via DORT here. Also mention that `onnxruntime` needs to be installed in addition to `onnx` and `onnxscript`.

This can be a separate PR though.

File: docs/source/onnx_torchscript.rst [hunk 5]
```diff
@@ -0,0 +1,716 @@
+TorchScript-based ONNX Exporter
+===============================
+
```
Comment (abock) [inline docs/source/onnx_torchscript.rst:3]: Should advertise the dynamo exporter here? Something like

> **NOTE**
> To export an ONNX model using TorchDynamo instead of TorchScript, see `torch.onnx.dynamo_export` (link to the new docs for dynamo export).

**Review by abock — CHANGES_REQUESTED** [review]
Looking nice! A few more suggestions...

File: docs/source/onnx_dynamo_onnxruntime_backend.rst (hunk 4 above)
Comment (thiagocrepaldi) [reply docs/source/onnx_dynamo_onnxruntime_backend.rst:10]: it is best to separate DORT and onnx exporter in separate documentation.
They should not be confused.

**Comment by thiagocrepaldi** [comment]
Superseeded by https://github.com/pytorch/pytorch/pull/109026
